tasks/configure.yml

```diff
@@ -37,12 +37,42 @@
 - name: PostgreSQL | Update configuration - pt. 2 (postgresql.conf)
   template:
     src: "postgresql.conf-{{ postgresql_version }}.j2"
-    dest: "{{postgresql_conf_directory}}/postgresql.conf"
+    # if using pgtune, save the template to ".untuned"
+    dest: "{{postgresql_conf_directory}}/postgresql.conf{% if postgresql_pgtune %}.untuned{% endif %}"
     owner: "{{ postgresql_service_user }}"
     group: "{{ postgresql_service_group }}"
     mode: 0640
   register: postgresql_configuration_pt2
 
+- name: PostgreSQL | Update configuration - pt. 3 (pgtune)
+  sudo: true
+  sudo_user: "{{ postgresql_service_user }}"
+  shell: |
+    TMPDIR=$(mktemp -d)
+    pgtune --input-config=postgresql.conf.untuned\
+           --output-config=$TMPDIR/postgresql.conf\
+           {% if postgresql_pgtune_memory %}--memory {{postgresql_pgtune_memory}}{% endif %}\
+           --type {{postgresql_pgtune_type }}\
+           {% if postgresql_pgtune_connections %}--type {{postgresql_pgtune_connections}}{% endif %}\
```

Comment:
This should be `--connections`, having `pgtune [...] type True` is causing the `pgtune` command to fail, but it fails silently because we are not checking its exit code, perhaps the `pgtune` task should run separately, then we can check its exit code? I had confused this as an idempotence issue earlier, but since digging into it more found the cause...

```bash
TASK: [PostgreSQL | Update configuration - pt. 3 (create pgtune conf)] ******** 
failed: [anxs.local] => {"changed": true, "cmd": "pgtune --input-config=postgresql.conf --output-config=/tmp/postgresql.conf.tuned --memory True --type Mixed --type True", "delta": "0:00:00.041031", "end": "2015-06-15 21:00:38.336155", "rc": 1, "start": "2015-06-15 21:00:38.295124", "warnings": []}
stdout: Error:  unexpected setting for dbType
```
Give me a shout when you've had time to look at this, happy to give it another lookover